server/heroku/processes.go

```diff
@@ -100,23 +99,19 @@ func (h *PostProcess) ServeHTTPContext(ctx context.Context, w http.ResponseWrite
 	}
 
 	if form.Attach {
-		inStream, outStream, err := hijackServer(w)
-		if err != nil {
-			return err
+		stream := &hijack.HijackReadWriter{
+			Response: w,
+			Header:   "HTTP/1.1 200 OK\r\nContent-Type: application/vnd.empire.raw-stream\r\n\r\n",
 		}
-		defer closeStreams(inStream, outStream)
-
-		fmt.Fprintf(outStream, "HTTP/1.1 200 OK\r\nContent-Type: application/vnd.empire.raw-stream\r\n\r\n")
-
+		defer stream.Close()
 		// Prevent the ELB idle connection timeout to close the connection.
-		defer close(streamhttp.Heartbeat(outStream, 10*time.Second))
+		defer close(streamhttp.Heartbeat(stream, 10*time.Second))
```

Comment:
i don't think we can avoid cases where we've already hijacked the connection and have started writing to it and then an error is returned from Run.

An example of this is if you try and run something that doesn't exist on the container:

```
$ ./build/emp run -a acme-inc worker -m test
Pulling from remind101/acme-inc
Digest: sha256:14d3e8043b5aae9fc730ae95774be993af5a0f7a16ef4852bde934b3da71fe0e
Status: Image is up to date for remind101/acme-inc:master
runner: start container: API error (500): Cannot start container 611e31671e0eb330a221adecc872935594c653fecfb78d445872039ad6a35f17: [8] System error: exec: "worker": executable file not found in $PATH
```

https://github.com/remind101/empire/pull/815/commits/7f751b44ed3efcc57f0b9d35ace44d0a5951be0b falls back to the existing behavior to just write the error to the stream and return `nil` if the connection has already been hijacked.

Given that, I'm not sure it's worth worry about the heartbeat and just allowing it to write the error to the stream.